de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/ResourceActivityQueuer.java

```diff
@@ -0,0 +1,85 @@
+package de.fu_berlin.inf.dpp.session.internal;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import de.fu_berlin.inf.dpp.activities.IActivity;
+import de.fu_berlin.inf.dpp.activities.IResourceActivity;
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.session.IActivityQueuer;
+
+/**
+ * This Resource based Queue gets a {@code Set} of {@link SPath}s, which should
+ * be queued and flushes them when the SPath entry was removed by
+ * {@link #disableQueuing(SPath)}.
+ * <p>
+ * To keep it simple, the execution order of queued activities stays unchanged.
+ * </p>
+ * TODO Discard obsolete Editor Events
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=156&issues=AWTIBaMkxXt_iLiSqxxk&open=AWTIBaMkxXt_iLiSqxxk)

[ISSUE_KEY]: AWTIBaMkxXt_iLiSqxxk